lib/sinon/collection.js

```diff
@@ -46,7 +46,16 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            if (typeof fake.resetHistory === "function") {
+                fake.resetHistory();
+                return;
+            }
+
+            if (typeof fake.reset === "function" && /^spy\#/.test(fake.id)) {
```

Comment:
None that I know of. Do you think I should just test for a `reset` method, and call it?